### PR TITLE
feat: 1252 Set document addressee 

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**
Update generate document views & adding addressee step

### CODE REVIEW

File: cases/services.py [hunk 1]
```diff
@@ -324,9 +324,8 @@ def post_generated_document(request, pk, json):
     return data.status_code
 
 
-def get_generated_document_preview(request, pk, tpk, text):
-    text = quote(text)
-    data = get(request, CASE_URL + pk + GENERATED_DOCUMENTS_PREVIEW_URL + "?template=" + tpk + "&text=" + text)
+def get_generated_document_preview(request, pk, params):
```
Comment (jfaracik) [inline cases/services.py:327]: Could you make these params explicit?

Comment (BenAndrew-Kainos) [reply cases/services.py:327]: I thought we had switched to making them non-explicit?

Comment (BenAndrew-Kainos) [reply cases/services.py:327]: Done

File: cases/forms/generate_document.py [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
+def select_template_form(templates, back_url):
     return Form(
         title=letter_templates.LetterTemplatesPage.PickTemplate.TITLE,
         questions=[
             RadioButtonsImage(
                 name="template",
                 options=[
                     Option(t["id"], t["name"], img_url=f"/assets/images/letter_templates/{t['layout']['filename']}.png")
-                    for t in templates
+                    for t in templates["results"]
```
Comment (rorysproule1) [inline cases/forms/generate_document.py:16]: Possible to add a clearer value than t?

Comment (BenAndrew-Kainos) [reply cases/forms/generate_document.py]: Good suggestion. Done

**Pull request merged**